controlplane/kubeadm/internal/workload_cluster_coredns.go

```diff
@@ -181,15 +183,21 @@ func (w *Workload) getCoreDNSInfo(ctx context.Context, clusterConfig *bootstrapv
 		return nil, err
 	}
 
+	// Handle the renaming of the upstream image from "k8s.gcr.io/coredns" to "k8s.gcr.io/coredns/coredns"
+	toImageName := parsedImage.Name
+	if toImageRepository == "k8s.gcr.io" && toImageName == "coredns" && targetMajorMinorPatch.GTE(semver.MustParse("1.8.0")) {
+		toImageName = "coredns/coredns"
```

Comment:
we should add some constants for these

Reply:
Similar to kubeadm does it, probably add a comment which references the kubeadm upstream contant here https://github.com/kubernetes/kubernetes/blob/release-1.21/cmd/kubeadm/app/constants/constants.go#L331?

Reply:
even better ^

Reply:
Constant added
I would avoid to link to the kubeadm code, because things can change there (e.g constant order was changed during v1.21); also, kubeadm is just adapting to what happen somewhere else